### PR TITLE
chore: upgrade to SDK v3

### PR DESCRIPTION
**Pull request by comcalvi**
Follow-up to #1706, which upgraded the node runtimes to 20. Node 20 is only compatible with SDK v3. This removes every use of the SDK v2.

-----

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: lib/change-control-lambda/disable-transition.ts [hunk 1]
```diff
@@ -1,6 +1,9 @@
 // eslint-disable-next-line import/no-extraneous-dependencies
-import * as AWS from 'aws-sdk';
-const pipeline = new AWS.CodePipeline();
+
+
+// eslint-disable-next-line import/no-extraneous-dependencies
+import { CodePipeline } from '@aws-sdk/client-codepipeline';
+const pipeline = new CodePipeline();
```
Comment (scanlonp) [inline lib/change-control-lambda/disable-transition.ts:6]:
```suggestion
import { CodePipeline } from '@aws-sdk/client-codepipeline';
const pipeline = new CodePipeline();
```
formatting clean up

File: lib/change-control-lambda/index.ts [hunk 2]
```diff
@@ -1,10 +1,13 @@
 // eslint-disable-next-line import/no-extraneous-dependencies
-import * as AWS from 'aws-sdk';
+
+
+// eslint-disable-next-line import/no-extraneous-dependencies
+import { S3 } from '@aws-sdk/client-s3';
```
Comment (scanlonp) [inline lib/change-control-lambda/index.ts:5]:
```suggestion
import { S3 } from '@aws-sdk/client-s3';
```
formatting clean up

File: lib/pipeline-watcher/handler/watcher-handler.ts [hunk 3]
```diff
@@ -1,5 +1,8 @@
 // eslint-disable-next-line import/no-extraneous-dependencies
-import * as AWS from 'aws-sdk';
+
+
+// eslint-disable-next-line import/no-extraneous-dependencies
+import { CloudWatch, Dimension, PutMetricDataCommandInput } from '@aws-sdk/client-cloudwatch';
```
Comment (scanlonp) [inline lib/pipeline-watcher/handler/watcher-handler.ts:5]:
```suggestion
import { CloudWatch, Dimension, PutMetricDataCommandInput } from '@aws-sdk/client-cloudwatch';
```
formatting clean up

**Review by scanlonp — APPROVED** [review]
A few nits. Otherwise looks good.

File: lib/change-control-lambda/disable-transition.ts (hunk 1 above)
Comment (rix0rrr) [reply lib/change-control-lambda/disable-transition.ts:6]: Why do you think the comment line `// eslint-disable-next-line import/no-extraneous-dependencies` is there?

**Pull request merged**